docs/v2_design.md

```diff
@@ -0,0 +1,179 @@
+# V2 Design Discussion
+
+**<big>Please help determining the based design for the library!</big>**
+
+---
+
+## Quick Intro
+
+Below is a short intro of what types are being currently generated in specific use cases.
+
+### Named quantity/dimension and unit
+
+```cpp
+auto p = 5 * si::power[W];
+
+static_assert(std::is_same_v<decltype(p), quantity<reference<isq::power_dim, si::watt>{}, int>>);
+```
+
+### Named quantity/dimension and derived (unnamed) unit
+
+```cpp
+auto s = 5 * si::speed[m / s];
+
+static_assert(std::is_same_v<decltype(s),
+              quantity<reference<isq::speed_dim, derived_unit<si::metre, per<si::second>>>{}, int>>);
+```
+
+### Derived (unnamed) quantity/dimension and derived (unnamed) unit
+
+```cpp
+auto q = 10 * si::length[m] / (2 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q),
+              quantity<reference<derived_dimension<isq::length_dim, per<isq::time_dim>>,
+                       derived_unit<si::metre, per<si::second>>>{}, int>>);
+```
+
+### Base quantity as a result of dimensional transformation
+
+```cpp
+auto d = 5 * si::speed[m / s] * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(d), quantity<reference<isq::length_dim, si::metre>{}, int>>);
+```
+
+### Dimensionless
+
+```cpp
+auto q = 20 * si::speed[m / s] / (10 * si::length[m]) * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q), quantity<reference<one_dim, one>{}, int>>);
+```
+
+---
+
+### Comparison
+
+- Should the below compile?
+- Should it be true?
+
+```cpp
+// Named and derived dimensions (same units)
+bool b1 = 10 * si::length[m] / (2 * si::time[s]) == 5 * si::speed[m / s];
+
+// Named and derived dimensions (different units)
+bool b2 = 10 / (2 * si::time[s]) == 5 * si::frequency[Hz];
+
+// Different named dimensions
+auto b3 = 5 * si::activity[Bq] == 5 * si::frequency[Hz];
+```
+
+---
+
+## Arithmetics
+
+- Should the below compile?
+- If yes, what should be the final type?
+
+```cpp
+// Named and derived dimensions (same units)
+auto q1 = 10 * si::length[m] / (2 * si::time[s]) + 5 * si::speed[m / s];
+
+// Named and derived dimensions (different units)
+auto q2 = 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+
+// Different named dimensions
+auto q3 = 5 * si::activity[Bq] + 5 * si::frequency[Hz];
+
+// ???
+auto q3a = 5 * si::activity[Bq] + 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+```
+
+---
+
+## Comparisons: Speed
+
+```cpp
+// same dimension type & different unit
+static_assert(1000 * si::length[m] == 1 * si::length[km]);
+// different dimension type (derived unnamed vs named)
+static_assert(20 * si::length[m] / (2 * si::time[s]) == 10 * si::speed[m]); // ???
```

Comment:
Being unable to mix these would be harmful to generic programming.
```C++
static_assert(square(x) == x.number() * si::area[m * m]); // `square` is generic.
```

Reply:
I am not sure what you mean here. Could you please provide more info?